doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst

```diff
@@ -60,6 +79,547 @@ Recommended changes
+              * between the main and the CHIP threads.
+              */
+              PlatformMgr().AddEventHandler(ChipEventHandler, 0);
+
+              err = PlatformMgr().StartEventLoopTask();
+              if (err != CHIP_NO_ERROR) {
+                LOG_ERR("PlatformMgr().StartEventLoopTask() failed");
+                return err;
+              }
+
+              return CHIP_NO_ERROR;
+
+         * Add the following code before the board components initialization.
+           The :c:func:`PrepareServer` method has two optional arguments that you can use to pass your own Matter event handler and  initialization data, including custom callbacks to invoke before and after the initialization.
+
+           .. code-block:: C++
+
+            /* Initialize Matter stack */
+	          ReturnErrorOnFailure(Nrf::Matter::PrepareServer(MatterEventHandler));
+
+         * Add the following code at the end of :c:func:`Init` method:
+
+           .. code-block:: C++
+
+            return Nrf::Matter::StartServer();
+
+      #. Add the ``main_init`` and ``matter_event_handler`` modules to the compilation.
+         Edit the :file:`CMakeLists.txt` file as follows:
+
+        .. code-block:: cmake
+
+          target_sources(app PRIVATE
+              ${COMMON_ROOT}/src/app/matter_init.cpp
+              ${COMMON_ROOT}/src/app/matter_event_handler.cpp
+          )
+
+    * Replace the code used for Matter event handling with the common ``matter_event_handler`` module.
+      The ``matter_event_handler`` module handles events generated by the Matter stack in a Nordic platform-specific way.
+      You can customize the module behavior by registering your own Matter event handler that extends the default implementation.
+
+      To replace the |NCS| v2.5.0 compliant implementation with the ``matter_event_handler`` module, complete the following steps:
+
+      1. Remove the :c:func:`MatterEventHandler` method declaration from the :file:`app_task.h` file.
+      #. Remove the :c:func:`MatterEventHandler` method implementation from the :file:`app_task.cpp` file.
+      #. Replace the code in the :c:func:`Init` method, in the :file:`app_task.cpp` file:
+
+         * Remove:
+
+           .. code-block:: C++
+
+            ReturnErrorOnFailure(Nrf::Matter::PrepareServer(MatterEventHandler));
+
+         * Add:
+
+           .. code-block:: C++
+
+            ReturnErrorOnFailure(Nrf::Matter::PrepareServer());
+
+            /* Register Matter event handler that controls the connectivity status LED based on the captured Matter network
+             * state. */
+             ReturnErrorOnFailure(Nrf::Matter::RegisterEventHandler(Nrf::Board::DefaultMatterEventHandler, 0));
+
+      #. Add the ``matter_event_handler`` module to the compilation.
+         Edit :file:`CMakeLists.txt` file as follows:
+
+         .. code-block:: cmake
+
+          target_sources(app PRIVATE
+              ${COMMON_ROOT}/src/app/matter_event_handler.cpp
+          )
+
```

Comment:
What about Replacing specific events for all samples? I mean, now there is a different way to manage application events. Some specific samples use it in a new way and customers may have used that specific sample as a template. For example, in the doorlock sample, the Identify handler uses Nrf::PostTask and Nrf::GetBoard to handle identifying, so when users migrate to the new API version they should replace it in this way within their application. And again the question is, do we want to describe here only the template sample? If so, then all things can stay as they are, otherwise I guess that we need to describe here the new PostTask method.

Reply:
Well, I think we can discuss it in the team. I assumed we don't want to create that detailed guide to mention all possible places in all samples that were changed. I selected template as an example, because it's the simplest one and it contains only generic code shared by all other samples. 
Regarding the fact that the new module can be used to post application events, it is mentioned in the line 95:

>       The ``board`` module handles buttons and LEDs in a way consistent with Matter samples UI.
>       It uses the ``task_executor`` common module for posting a board related events.
>       You can also use the ``task_executor`` module for posting and dispatching events in your application.